packages/application-shell/src/components/custom-view-dev-host/custom-view-dev-host.tsx

```diff
@@ -0,0 +1,148 @@
+  environment: TApplicationContext<{}>['environment'];
+};
+
+const LocalCustomViewLauncher = (props: TLocalCustomViewLauncherProps) => {
+  const [shouldRenderCustomView, setShouldRenderCustomView] = useState(false);
+  const [hostUrl, setHostUrl] = useState('');
+
+  const customViewConfig = getCustomViewConfig({
+    type: props.environment.__DEVELOPMENT__?.customViewType as TCustomViewType,
+    typeSettings: props.environment.__DEVELOPMENT__?.customViewTypeSettings,
+  });
+
+  return (
+    <main>
+      <Constraints.Horizontal max="scale">
+        <Spacings.Inset scale="xl">
+          <Spacings.Stack scale="xl">
+            <Text.Headline as="h1">Custom View loader</Text.Headline>
+
+            <Spacings.Stack scale="l">
+              <Spacings.Stack scale="m">
+                <Text.Body>
+                  This page simulates the Merchant Center context so you can run
+                  your Custom View locally.
+                  <br />
+                  Based on your <i>custom-view-config</i> file, these are the
+                  settings that will be used:
+                </Text.Body>
+                <ul>
+                  <li>
+                    Custom View type: <b>{customViewConfig.type}</b>
+                  </li>
+                  <li>
+                    Custom View size:{' '}
+                    <b>{customViewConfig.typeSettings?.size}</b>
+                  </li>
+                </ul>
+              </Spacings.Stack>
+              <Constraints.Horizontal max={10}>
+                <Spacings.Stack scale="l">
+                  <TextField
+                    title="Host URL"
+                    description="The Custom View will receive the URL of the Merchant Center current page in case you need to load data based on it. Populate this field in case you want to simulate a specific URL locally."
+                    placeholder="/<application_name>/<project_key>/<view_path>"
+                    value={hostUrl}
+                    onChange={(event) => setHostUrl(event.target.value)}
+                  />
+                  <PrimaryButton
+                    label="Open the Custom View"
+                    onClick={() => setShouldRenderCustomView(true)}
+                  />
+                </Spacings.Stack>
+              </Constraints.Horizontal>
+            </Spacings.Stack>
+          </Spacings.Stack>
+
+          {shouldRenderCustomView && (
+            <CustomViewLoader
+              customView={customViewConfig}
+              hostUrl={hostUrl}
+              onClose={() => setShouldRenderCustomView(false)}
+            />
+          )}
+        </Spacings.Inset>
+      </Constraints.Horizontal>
+    </main>
+  );
+};
+
+type TCustomViewDevHost = {
+  environment: TApplicationContext<{}>['environment'];
+  applicationMessages: TAsyncLocaleDataProps['applicationMessages'];
+  children: ReactNode;
+};
+
+const CustomViewDevHost = (props: TCustomViewDevHost) => {
```

Comment:
This component is used as the landing page for Custom Views local development so users can use a Custom Application like host to trigger rendering the Custom View as it will be done in the Merchant Center.

![image](https://github.com/commercetools/merchant-center-application-kit/assets/97907068/ac77b090-df5d-4ce6-a91d-ac2cee139031)

![image](https://github.com/commercetools/merchant-center-application-kit/assets/97907068/7af8ed06-5269-4f16-92f6-3fc119854d71)